api/v1beta2/gitrepository_types.go

```diff
@@ -97,8 +97,11 @@ type GitRepositorySpec struct {
 
 	// GitImplementation specifies which Git client library implementation to
 	// use. Defaults to 'go-git', valid values are ('go-git', 'libgit2').
+	// Deprecated: gitImplementation is deprecated now that 'go-git' is the
+	// only supported implementation.
 	// +kubebuilder:validation:Enum=go-git;libgit2
 	// +kubebuilder:default:=go-git
+	// +kubebuilder:deprecatedversion:warning="gitImplementation is deprecated"
```

Comment:
Not sure if this is OK, we set a default for this field so it can't be unset resulting in forever warnings in logs, kubectl and Flux CLI output.